e2e/playwright/actions/src/tests/share/share-file.spec.ts

```diff
@@ -0,0 +1,246 @@
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiClientFactory, NodesApi, test, timeouts, Utils, SharedLinksApi } from '@alfresco/playwright-shared';
+import { expect } from '@playwright/test';
+
+test.describe('Share a file', () => {
+  const random = Utils.random();
+
+  const username = `user-${random}`;
+  const parent = `parent-${random}`;
+  let parentId: string;
+
+  const file3 = `file3-${random}.txt`;
+  const file4 = `file4-${random}.txt`;
+  const file5 = `file5-${random}.txt`;
+  const file6 = `file6-${random}.txt`;
+  const file7 = `file7-${random}.txt`;
+  const file8 = `file8-${random}.txt`;
+  const file9 = `file9-${random}.txt`;
+
+  const shareLinkPreUrl = `/#/preview/s/`;
+
+  const apiClientFactory = new ApiClientFactory();
+
+  test.beforeAll(async () => {
+    await apiClientFactory.setUpAcaBackend('admin');
+    await apiClientFactory.createUser({ username });
+    const nodesApi = await NodesApi.initialize(username, username);
+
+    parentId = (await nodesApi.createFolder(parent)).entry.id;
+  });
+
+  test.afterAll(async () => {
+    await apiClientFactory.nodes.deleteNodes([parentId]);
+  });
+
+  test.describe('when logged out', () => {
+    let file6SharedLink: string;
+    let file6Id: string;
+
+    test.beforeAll(async () => {
+      const nodesApi = await NodesApi.initialize(username, username);
+      const shareApi = await SharedLinksApi.initialize(username, username);
+
+      file6Id = (await nodesApi.createFile(file6, parentId))?.entry.id;
+
+      const sharedId = (await shareApi.shareFileById(file6Id)).entry.id;
+      file6SharedLink = `${shareLinkPreUrl}${sharedId}`;
+      await shareApi.waitForFilesToBeShared([file6Id]);
+    });
+
+    test.afterAll(async () => {
+      await apiClientFactory.nodes.deleteNodes([file6Id]);
+    });
+
+    test('[C286326] A non-logged user can download the shared file from the viewer', async ({ personalFiles, page }) => {
+      await page.goto(file6SharedLink);
+      await personalFiles.viewer.waitForViewerToOpen();
+
+      const downloadPromise = personalFiles.page.waitForEvent('download');
+      await personalFiles.viewer.toolbar.sharedDownloadButton.click();
+      const download = await downloadPromise;
+      expect(download.suggestedFilename()).toBe(file6);
+    });
+  });
+
+  test.describe('when logged in', () => {
+    const expiryDateObj: Date = new Date();
+    expiryDateObj.setFullYear(expiryDateObj.getFullYear() + 1);
+    const expiryDate: any = expiryDateObj.toISOString().replace('Z', '+0000');
```

Comment:
Please don't use `any`

Reply:
removing 'any' here is giving a red line error on _expiryDate_ here 
`await shareApi.shareFilesByIds([file6Id, file7Id], expiryDate);`
because _shareFileByIds()_ and other nested methods expect _expireDate_ to be of type 'Date'. But strangely, the _shareFilesByIds()_ method works only if the _expireDate_ is of type 'string', otherwise the tests don't work as expected.
![image](https://github.com/Alfresco/alfresco-content-app/assets/110394264/e08c57bd-74f6-498d-a9c5-db59418af9be)


Due to this strange behavior, having 'any' helps but if I remove 'any', I will have to modify the data type of _expireDate_ from 'Date' to 'string' at multiple places and in JS-API which will need upstream.

I have tried multiple ways to make _expiryDate_ an object of type 'Date' so that no changes are required on the API level but that does not seem to work

What do you suggest here? Should we go ahead with 'any'? or can we use some other approach here? @MichalKinas

Reply:
If you need to pass a `Date` there you can use `expiryDateObj` object, `expiryDate` should be a string and using `any` here is incorrect

Reply:
I have tried using expiryDateObj but this does not help as 'shareFilesByIds()' is working properly only with expiryDate as string. 
If we remove any, we get an error because 'shareFilesByIds()' expects a Date but is somehow working only with string

Reply:
You can try creating new date from `expiryDate` but if it won't work it means that your code is not correct somewhere, `any` is anti-pattern and it should be used only in specifi cases and this is not one of them. `any` shouldn't be used to fix an error or not working code

Reply:
Yes, I understand what you are saying but the problem I am facing in doing so is that
1. the API actually expects 'expiryDate' to be of type string. (see sharedLinkBodyCreate https://api-explorer.alfresco.com/api-explorer/?urls.primaryName=Core%20API#/shared-links/createSharedLink)
![image](https://github.com/Alfresco/alfresco-content-app/assets/110394264/02a6a4c0-0491-442c-9ccc-f103f32b49d4)


2. But, in the code, the expected type of 'expiryDate' is set to Date in JS API and all related methods like 'shareNodes()' or 'shareFileByIds()'
![image](https://github.com/Alfresco/alfresco-content-app/assets/110394264/6f1974c7-8d4e-470d-946c-3980f5db6721)


So even if I create new date from 'expiryDate' , the API will still not run as expected. 
So far, this approach (using 'any') was being used in Protractor e2es so if we remove 'any', we either get some red line errors but the tests run fine 
OR 
we might need to make changes in JS API which I am not sure is feasible.

Reply:
In this case you still don't have to use `any`, you can declare `expiryDate` as `let expiryDate: Date | string = ...` and then pass it to the correct function like `expiryDate as Date`

Reply:
Passing expiryDate as Date does not help because `this.apiService.share.createSharedLink(data);` fails if expiry is of type 'Date' and gives status code 400.
It gives 200 only with type string (as per swagger https://api-explorer.alfresco.com/api-explorer/?urls.primaryName=Core%20API#/shared-links/createSharedLink)
and I am blocked because I cannot set the type of expiryDate to string. 
![image (3)](https://github.com/Alfresco/alfresco-content-app/assets/110394264/2b7915d7-fbc9-4cb2-a743-1fb811a32d67)
![image (2)](https://github.com/Alfresco/alfresco-content-app/assets/110394264/98d60231-6890-4209-99f6-ca2c1fb96add)

@DenysVuika, Michal suggested double-checking with you if we should make changes for this in JS-API for e2es. 
Can you please suggest what would be the feasible solution?

Reply:
@MichalKinas, as per my discussion with @DenysVuika, we can use 'any' for now and have a separate ticket to fix typings in JS-API. 

I have created this JIRA https://alfresco.atlassian.net/browse/ACS-6533 for the same.